### PR TITLE
add in bango to solitude-auth

### PR DESCRIPTION
**Pull request by andymckay**
for mozilla/solitude#444

### CODE REVIEW

File: auth/settings/base.py [hunk 1]
```diff
@@ -79,21 +94,23 @@
...
-BANGO_USERNAME = ''
-BANGO_PASSWORD = ''
+BANGO_AUTH = {
+    'USER': os.environ.get('BANGO_AUTH_USER'),
+    'PASSWORD': os.environ.get('BANGO_AUTH_PASSWORD')
+}
 
-BRAINTREE_PUBLIC_KEY = ''
-BRAINTREE_PRIVATE_KEY = ''
+BRAINTREE_PUBLIC_KEY = os.environ.get('BRAINRTEE_PUBLIC_KEY')
```
Comment (kumar303) [inline auth/settings/base.py]: type: BRAINRTEE :fried_shrimp:

File: auth/tests/test.py [hunk 2]
```diff
@@ -22,3 +22,35 @@ def test_ok(self):
...
+        </ns0:CreatePackage>
+    </ns1:Body>
+</SOAP-ENV:Envelope>"""  # noqa
+
+
+@override_settings(BANGO_AUTH={'USER': 'bango', 'PASSWORD': 'bob'})
+class TestBango(BaseTestCase):
+
+    def setUp(self):
+        super(self.__class__, self).setUp()
```
Comment (kumar303) [inline auth/tests/test.py]: I keep forgetting to use ``self.__class__``

File: auth/utils.py [hunk 3]
```diff
@@ -33,13 +34,14 @@ def prepare(request):
...
+    for incoming, outgoing in settings.HEADERS_ALLOWED.items():
         # Transform the key from the settings into the appropriate
         # format from Django request.
-        meta = 'HTTP_' + k.upper().replace('-', '_')
+        meta = incoming.upper().replace('-', '_')
         if meta in request.META:
-            new['headers'][k] = request.META[meta]
-            log.info('Adding header: {0}, {1}'.format(k, request.META[meta]))
+            new['headers'][outgoing] = request.META[meta]
+            log.info('Adding header: {0}, {1}'
```
Comment (kumar303) [inline auth/utils.py]: maybe also log the outgoing header key?

File: auth/tests/test.py [hunk 4]
```diff
@@ -22,3 +22,35 @@ def test_ok(self):
...
+        self.url = reverse('bango')
+
+    def test_ok(self):
+        self.req.post.return_value = self.get_response('foo', 200)
+        assert self.post(self.url, example).status_code, 200
+        data = example.replace('weee!', 'bango').replace('wooo!', 'bob')
+        self.req.post.assert_called_with(
+            'https://b.c/some/url/', verify=True,
+            data=data, timeout=30, headers={'Content-Type': 'text/xml'}
```
Comment (kumar303) [inline auth/tests/test.py]: might be more flexible as ``timeout=settings.DEFAULT_TIMEOUT``

File: Dockerfile [hunk 5]
```diff
@@ -6,9 +6,8 @@ FROM mozillamarketplace/centos-mysql-mkt:0.2
 
 RUN yum install -y supervisor bash-completion && yum clean all
```
Comment (kumar303) [inline Dockerfile]: don't we need libxslt and friend for lxml?

**Comment by kumar303** [comment]
r+wc

File: auth/settings/base.py (hunk 1 above)
Comment (andymckay) [inline auth/settings/base.py]: thanks actually got that already, need to rebase

File: auth/utils.py (hunk 3 above)
Comment (andymckay) [inline auth/utils.py]: doh, it makes more sense to log outgoing than incoming

File: auth/tests/test.py (hunk 4 above)
Comment (andymckay) [inline auth/tests/test.py]: sure

File: Dockerfile (hunk 5 above)
Comment (andymckay) [inline Dockerfile]: Its in a higher image already:
```
solitude docker history mozillapayments/solitude-auth | grep libxslt                        verify-on-auth  ✱ pay
ceaa1c83f483        9 months ago        /bin/sh -c yum install -y libxslt               18.13 MB
```

**Pull request merged**